File: sdk/nodejs/src/resources/will_expired.js
"use strict";

const { BaseResource } = require("./base");

class WillExpiredResource extends BaseResource {
  async list(params = {}) {
    return this._http.get("/api/willexpired");
  }

  async getSummary(params = {}) {
    return this._http.get("/api/willexpired/summary");
  }

  async getConfig() {
    return this._http.get("/api/willexpired/config");
  }

  async listInvoices() {
    return this._http.get("/api/willexpired/invoices");
  }

  async listRequests() {
    return this._http.get("/api/willexpired/requests");
  }

  async exportAll() {
    return this._http.get("/api/willexpired/export");
  }

  async getItem(itemType, itemId) {
    return this._http.get("/api/willexpired/{type}/{id}");
  }

  async renew(itemType, itemId) {
    return this._http.post("/api/willexpired/{type}/{id}/renew");
  }

  async getAutorenew(itemType, itemId) {
    return this._http.get("/api/willexpired/{type}/{id}/autorenew");
  }

  async setAutorenew(itemType, itemId, data) {
    return this._http.put("/api/willexpired/{type}/{id}/autorenew", data);
  }

}

module.exports = { WillExpiredResource };

File: sdk/csharp/Resources/WillExpired.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HiTechCloud.SDK.Resources;

public class WillExpired : BaseResource
{
    public WillExpired(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> ListAsync()
    {
        return await Http.GetAsync("/api/willexpired");
    }

    public async Task<Dictionary<string, object?>> SummaryAsync()
    {
        return await Http.GetAsync("/api/willexpired/summary");
    }

    public async Task<Dictionary<string, object?>> ConfigAsync()
    {
        return await Http.GetAsync("/api/willexpired/config");
    }

    public async Task<Dictionary<string, object?>> InvoicesAsync()
    {
        return await Http.GetAsync("/api/willexpired/invoices");
    }

    public async Task<Dictionary<string, object?>> RequestsAsync()
    {
        return await Http.GetAsync("/api/willexpired/requests");
    }

    public async Task<Dictionary<string, object?>> ExportAsync()
    {
        return await Http.GetAsync("/api/willexpired/export");
    }

    public async Task<Dictionary<string, object?>> ItemAsync(string type, int id)
    {
        return await Http.GetAsync(string.Format("/api/willexpired/{type}/{id}", type, id));
    }

    public async Task<Dictionary<string, object?>> RenewAsync(string type, int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/willexpired/{type}/{id}/renew", type, id), data);
    }

    public async Task<Dictionary<string, object?>> AutorenewAsync(string type, int id)
    {
        return await Http.GetAsync(string.Format("/api/willexpired/{type}/{id}/autorenew", type, id));
    }

    public async Task<Dictionary<string, object?>> SetAutorenewAsync(string type, int id, Dictionary<string, object> data)
    {
        return await Http.PutAsync(string.Format("/api/willexpired/{type}/{id}/autorenew", type, id), data);
    }
}
